### PR TITLE
Gjoranv/New metrics proxy

### PR DESCRIPTION
**Pull request by gjoranv**
FYI: @bratseth

### CODE REVIEW

**Review by hmusum — APPROVED** [review]
LGTM, although it's  hard to review the changes in the first commit, since code was moved to this repo and modified in the same commit.

File: config-model/src/main/java/com/yahoo/searchdefinition/FeatureNames.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 /*
- * // Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+* Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/searchdefinition/FeatureNames.java]: Duplicate copyright

File: metrics-proxy/pom.xml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        <groupId>com.yahoo.vespa</groupId>
+        <artifactId>bundle-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <Bundle-SymbolicName>${project.groupId}.${project.artifactId}</Bundle-SymbolicName>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <!-- Only added to make IntelliJ use correct language level -->
```
Comment (bjorncs) [inline metrics-proxy/pom.xml]: Groupid is not required?

**Review by bjorncs — APPROVED** [review]

File: metrics-proxy/src/test/java/ai/vespa/metricsproxy/service/ContainerServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * @author Unknown
+ */
+public class ContainerServiceTest {
+
+    private MockHttpServer service;
+    private int csPort;
+    private static final String response;
+
+    static {
+        response = TestUtil.getContents(new File("src/test/resources/metrics-container-state-multi-chain.json"));
```
Comment (bjorncs) [inline metrics-proxy/src/test/java/ai/vespa/metricsproxy/service/ContainerServiceTest.java]: This should use current classloader to correctly resolve path to test resource.

File: metrics-proxy/src/test/java/ai/vespa/metricsproxy/service/ContainerServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * @author Unknown
+ */
+public class ContainerServiceTest {
+
+    private MockHttpServer service;
+    private int csPort;
+    private static final String response;
+
+    static {
```
Comment (bjorncs) [inline metrics-proxy/src/test/java/ai/vespa/metricsproxy/service/ContainerServiceTest.java]: Should ideally use `@BeforeClass`

**Review by bjorncs — APPROVED** [review]

File: metrics-proxy/pom.xml (hunk 2 above)
Comment (gjoranv) [reply metrics-proxy/pom.xml]: No, I think it's optional for the native plugins.

**Comment by gjoranv** [comment]
@bjorncs, resolved all your comments. PTAL.

**Review by bjorncs — APPROVED** [review]

**Review by bjorncs — COMMENTED** [review]
👍

**Pull request merged**